skfp/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,224 @@
+    validate_train_valid_test_split_sizes,
+)
+from skfp.utils.validators import ensure_mols
+
+
+@validate_params(
+    {
+        "data": ["array-like"],
+        "additional_data": ["tuple"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "test_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "include_chirality": ["boolean"],
+        "return_indices": ["boolean"],
+    },
+    prefer_skip_nested_validation=True,
+)
+def maxmin_train_test_split(
+    data: Sequence[Union[str, Mol]],
+    *additional_data: Sequence,
+    train_size: Optional[float] = None,
+    test_size: Optional[float] = None,
+    return_indices: bool = False,
+    random_state: int = 0,
+) -> Union[
+    tuple[
+        Sequence[Union[str, Mol]], Sequence[Union[str, Mol]], Sequence[Sequence[Any]]
+    ],
+    tuple[Sequence, ...],
+    tuple[Sequence[int], Sequence[int]],
+]:
+    """
+    Split using maxmin algorithm or picking a subset of item from a pool.
+    Implements specific
+
+    Parameters
+    ----------
+    data : sequence
+        A sequence representing either SMILES strings or RDKit `Mol` objects.
+
+    additional_data: list[sequence]
+        Additional sequences to be split alongside the main data (e.g., labels or feature vectors).
+
+    train_size : float, default=None
+        The fraction of data to be used for the train subset. If None, it is set to 1 - test_size.
+        If test_size is also None, it will be set to 0.8.
+
+    test_size : float, default=None
+        The fraction of data to be used for the test subset. If None, it is set to 1 - train_size.
+        If train_size is also None, it will be set to 0.2.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit `Mol` objects, or only the indices of the subsets instead of the data.
+
+    Returns
+    ----------
+    subsets : tuple[list, list, ...]
+    Tuple with train-test subsets of provided arrays. First two are lists of SMILES strings or RDKit `Mol` objects,
+    depending on the input type. If `return_indices` is True, lists of indices are returned instead of actual data.
+    References
+    ----------
+    https://github.com/deepchem/deepchem
+    https://rdkit.org/docs/cppapi/classRDPickers_1_1MaxMinPicker.html
+    https://squonk.it/docs/cells/RDKit%20MaxMin%20Picker/
+    https://rdkit.blogspot.com/2017/11/revisting-maxminpicker.html
+
+    """
+    data_size = len(data)
+    train_size, test_size = validate_train_test_split_sizes(
+        train_size, test_size, data_size
+    )
+
+    molecules = ensure_mols(data)
+    fingerprints = [
+        AllChem.GetMorganFingerprintAsBitVect(mol, 2, 1024) for mol in molecules
+    ]
```

Comment:
1. Add arguments `fingerprints` with default value None. If only they are not provided, calculate this
2. Use `ECFPFingerprint` instead of raw RDKit

Reply:
1. Done
2. Lets talk about it on weekly, I thought rawRDKit with radious > 1, is like ECFPFingerprint

Reply:
It is, so why not use it? We have our own classes for convenience, after all, and it's shorter:
```
fps = ECFPFingerprint().transform(molecules)
```
Also it uses more standard 2048 bits by default.